realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -396,15 +402,17 @@ private static void initializeSyncedRealm(Realm realm) {
 
             // Assumption: When SyncConfiguration then additive schema update mode.
             final OsRealmSchema schema = new OsRealmSchema(schemaCreator);
-            long newVersion = realm.configuration.getSchemaVersion();
+            long newVersion = configuration.getSchemaVersion();
```

Comment:
@beeender : I don't know!  The code didn't close it, before.  Do you think it needs to do so, now?